### PR TITLE
Improve and unify the logging framework

### PR DESCRIPTION
**Pull request by meenbeese**
This PR enhances our discombobulated logging framework by unifying all logs through our custom logger utility, replacing the use of `print` and `debugPrint` methods as suggested by Flutter. I have updated the `logger` library to the latest version and removed the unused `logging` library from our dependencies. These changes streamline debugging across various packages, methods, and classes for improved efficiency.

### CODE REVIEW

**Comment by meenbeese** [comment]
I had some issues with conflicting commits and rebases but they should be fixed now.

File: lib/main.dart [hunk 1]
```diff
@@ -35,8 +35,8 @@ import 'package:flutter_native_splash/flutter_native_splash.dart';
 import 'package:flutter_displaymode/flutter_displaymode.dart';
 
 Future<void> main(List<String> rawArgs) async {
+  final logger = getLogger("Spotube");
```
Comment (KRTirtho) [inline lib/main.dart:38]: Why do we need it here?

File: bin/gen-credits.dart [hunk 2]
```diff
@@ -6,8 +6,11 @@ import 'package:http/http.dart';
 import 'package:html/parser.dart';
 import 'package:pub_api_client/pub_api_client.dart';
 import 'package:pubspec_parse/pubspec_parse.dart';
+import 'package:spotube/models/logger.dart';
 
 void main() async {
+  final logger = getLogger("GenCredits");
```
Comment (KRTirtho) [inline bin/gen-credits.dart:12]: Please revert this file's changes. It's not for debug purposes. The output needs not to be decorated

File: bin/untranslated_messages.dart [hunk 3]
```diff
@@ -13,6 +14,8 @@ import 'dart:io';
 /// Example: dart bin/untranslated_messages.dart bn
 
 void main(List<String> args) {
+  final logger = getLogger("UntranslatedMessages");
```
Comment (KRTirtho) [inline bin/untranslated_messages.dart:17]: Revert this one as well. We need clean logs not decorated logs

File: bin/verify-pkgbuild.dart [hunk 4]
```diff
@@ -15,8 +17,7 @@ void main() {
         throw Exception("In new releases pkgrel should be 1");
       }
     } catch (e) {
-      // ignore: avoid_print
-      print("[Failed to parse PKGBUILD] $e");
+      logger.d("[Failed to parse PKGBUILD] $e");
```
Comment (KRTirtho) [inline bin/verify-pkgbuild.dart:20]: This should be `logger.e`

File: lib/provider/proxy_playlist/next_fetcher_mixin.dart [hunk 5]
```diff
@@ -123,8 +126,8 @@ mixin NextFetcher on StateNotifier<ProxyPlaylist> {
         );
       }
     } catch (e, stackTrace) {
-      debugPrint(e.toString());
-      debugPrintStack(stackTrace: stackTrace);
+      logger.d(e.toString());
```
Comment (KRTirtho) [inline lib/provider/proxy_playlist/next_fetcher_mixin.dart:129]: `logger.e` should be used

**Review by KRTirtho — CHANGES_REQUESTED** [review]

**Comment by meenbeese** [comment]
Messed up this PR very badly, opening a new one...

**Pull request closed without merging**